ERCS/erc-7656.md

```diff
@@ -0,0 +1,279 @@
+---
+eip: 7656
+title: Generalized Token-Linked Contracts
+description: Define a registry for generic token-bound contracts
+author: Francesco Sullo (@sullof)
+discussions-to: https://ethereum-magicians.org/t/variation-to-erc6551-to-deploy-any-kind-of-contract-linked-to-an-nft/19223
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-03-15
+requires: 165, 6551
+---
+
+## Abstract
+
+This proposal introduces a variation of [ERC-6551](./eip-6551.md) that extends to all types of contracts linked to non-fungible tokens (NFTs). This generalization allows NFTs not only to own assets and interact with applications as accounts but also to be linked with any contract, enhancing their utility without necessitating modifications to existing smart contracts or infrastructure.
+
+## Motivation
+
+Our initial approach involved proposing an expansion of [ERC-6551](./eip-6551.md) to encompass a broader scope of token-bound contracts beyond accounts. The goal was to enable any deployed token-bound contract to potentially represent more than just an account, with the actual functionality determined by checking the contract's interface. Unfortunately, this suggestion was not adopted due to concerns about complicating the trust model built into [ERC-6551](./eip-6551.md), where projects rely on emitted events to understand the nature of the contract without additional verification steps. Our proposed change, while introducing versatility, would have necessitated interface checks by developers, introducing a layer of friction and potentially leading to non-account contracts being disregarded as irrelevant by Token Bound Account (TBA) indexes.
+
+In developing a protocol aimed at enriching the functionalities of NFTs, we encountered the need to deploy a dedicated manager and various plugins for each NFT to expand its capabilities. While some plugins functioned as accounts, mirroring the [ERC-6551](./eip-6551.md) approach, we found that the majority did not fit within this framework. This discrepancy necessitated the creation of a new, more flexible registry that could accommodate a wider range of functionalities. Initially, this registry was intended solely for our internal use to overcome the limitations we faced. However, as our work progressed, we recognized that the challenges we encountered were not unique to our project but were indicative of a broader need within the ecosystem for greater flexibility in linking NFTs with diverse contract types. Inspired by this insight and encouraged by preliminary feedback from the community, we decided to formalize our solution and propose it as an ERC, with the belief that it could significantly benefit the entire Ethereum ecosystem by providing a versatile framework for NFT functionalities.
```

Comment:
While this is good background information, the Motivation section should convince the reader that this proposal is the best for the problem they are trying to solve.

Reply:
Ok. I will leave that only in the discussion so that at least there people can get the reasons why we make this change.